modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/notifications/CustomRXTStoreNotificationTestCase.java

```diff
@@ -0,0 +1,232 @@
+
+    @Factory(dataProvider = "userModeProvider")
+    public CustomRXTStoreNotificationTestCase(TestUserMode userMode) {
+        this.userMode = userMode;
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init(userMode);
+        String session = getSessionCookie();
+        genericRestClient = new GenericRestClient();
+        queryParamMap = new HashMap<String, String>();
+        headerMap = new HashMap<String, String>();
+        resourcePath =
+                FrameworkPathUtil.getSystemResourceLocation() + "artifacts" + File.separator + "GREG" + File.separator;
+        publisherUrl = automationContext.getContextUrls().getSecureServiceUrl().replace("services", "publisher/apis");
+        storeUrl = automationContext.getContextUrls().getSecureServiceUrl().replace("services", "store/apis");
+        resourceAdminServiceClient = new ResourceAdminServiceClient(backendURL, session);
+        addCustomRxt();
+        setTestEnvironment();
+    }
+
+    @Test(groups = { "wso2.greg",
+            "wso2.greg.es" }, description = "Adding subscription to custom asset on LC state change")
+    public void addSubscriptionToLcStateChange() throws JSONException, IOException {
+        JSONObject dataObject = new JSONObject();
+        dataObject.put("notificationType", "StoreLifeCycleStateChanged");
+        dataObject.put("notificationMethod", "work");
+
+        ClientResponse response = genericRestClient
+                .geneticRestRequestPost(storeUrl + "/subscription/applications/" + assetId, MediaType.APPLICATION_JSON,
+                        MediaType.APPLICATION_JSON, dataObject.toString(), queryParamMap, headerMap, cookieHeaderStore);
+
+        String payLoad = response.getEntity(String.class);
+        payLoad = payLoad.substring(payLoad.indexOf('{'));
+        JSONObject obj = new JSONObject(payLoad);
+        assertNotNull(obj.get("id").toString(),
+                "Response payload is not the in the correct format" + response.getEntity(String.class));
+
+        genericRestClient.geneticRestRequestPost(publisherUrl + "/assets/" + assetId + "/state",
+                MediaType.APPLICATION_FORM_URLENCODED, MediaType.APPLICATION_JSON,
+                "nextState=Testing&comment=Completed", queryParamMap, headerMap, cookieHeaderPublisher);
+        // TODO - Since notification not appearing in the store
+    }
+
+    @Test(groups = { "wso2.greg",
+            "wso2.greg.es" }, description = "Adding subscription to custom asset on resource update")
+    public void addSubscriptionToResourceUpdate() throws JSONException, IOException {
+        JSONObject dataObject = new JSONObject();
+        dataObject.put("notificationType", "StoreResourceUpdated");
+        dataObject.put("notificationMethod", "work");
+
+        ClientResponse response = genericRestClient
+                .geneticRestRequestPost(storeUrl + "/subscription/applications/" + assetId, MediaType.APPLICATION_JSON,
```

Comment:
This might not be your code. But there is a typo in genericRestClient.geneticRestRequestPost. (genetic should be generic, right?)